indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -793,13 +793,17 @@ public void onFailure(Throwable t)
         status = Status.PUBLISHING;
       }
 
+      // The callback for a successful segment publish may remove a sequence from the publishingSequences,
+      // which is racy and can allow the same sequence to be added to the set again.
+      // Create a copy of publishing sequences to which we can only add elements, and not remove them.
+      final Set<String> publishingSequencesSnapshot = new HashSet<>(publishingSequences);
```

Comment:
> Also, I wonder if there can still be duplicates as it really depends on the timing of snapshotting publishingSequences. Would a better solution be to track publishedSequences?

I think that this may not happen since the removal happens from sequences and then from publishingSequences, so this order may help. Although it doesn't seem ideal.

Thanks, I'll try to track publishedSequences instead.